app/upgrades/v17/upgrades_test.go

```diff
@@ -125,10 +142,50 @@ func (suite *UpgradeTestSuite) TestUpgrade() {
 					lastPoolID = poolID
 				}
 
-				return expectedCoinsUsedInUpgradeHandler, lastPoolID
+				existingPool := suite.PrepareConcentratedPoolWithCoins("ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo")
+				existingPool2 := suite.PrepareConcentratedPoolWithCoins("akash", "uosmo")
+
+				// create few TWAP records for the pools
+				t1 := dummyTwapRecord(existingPool.GetId(), time.Now().Add(-time.Hour*24), "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo", sdk.NewDec(10),
+					sdk.OneDec().MulInt64(10*10),
+					sdk.OneDec().MulInt64(3),
+					sdk.ZeroDec())
+
+				t2 := dummyTwapRecord(existingPool.GetId(), time.Now().Add(-time.Hour*10), "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo", sdk.NewDec(30),
+					sdk.OneDec().MulInt64(10*10+10),
+					sdk.OneDec().MulInt64(5),
+					sdk.ZeroDec())
+
+				t3 := dummyTwapRecord(existingPool.GetId(), time.Now().Add(-time.Hour), "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", "uosmo", sdk.NewDec(20),
+					sdk.OneDec().MulInt64(10*10+10*5),
+					sdk.OneDec().MulInt64(10),
+					sdk.ZeroDec())
+
+				t4 := dummyTwapRecord(existingPool2.GetId(), time.Now().Add(-time.Hour*24), "akash", "uosmo", sdk.NewDec(10),
+					sdk.OneDec().MulInt64(10*10*10),
+					sdk.OneDec().MulInt64(5),
+					sdk.ZeroDec())
+
+				t5 := dummyTwapRecord(existingPool2.GetId(), time.Now().Add(-time.Hour), "akash", "uosmo", sdk.NewDec(20),
+					sdk.OneDec().MulInt64(10),
+					sdk.OneDec().MulInt64(2),
+					sdk.ZeroDec())
+
+				suite.App.TwapKeeper.StoreNewRecord(suite.Ctx, t1)
+				suite.App.TwapKeeper.StoreNewRecord(suite.Ctx, t2)
+				suite.App.TwapKeeper.StoreNewRecord(suite.Ctx, t3)
+				suite.App.TwapKeeper.StoreNewRecord(suite.Ctx, t4)
+				suite.App.TwapKeeper.StoreNewRecord(suite.Ctx, t5)
+
+				clPool2TwapRecordPreUpgrade, err := keepers.TwapKeeper.GetAllMostRecentRecordsForPool(ctx, existingPool2.GetId())
+				suite.Require().NoError(err)
+
+				return expectedCoinsUsedInUpgradeHandler, lastPoolID, clPool2TwapRecordPreUpgrade
 
 			},
-			func(ctx sdk.Context, keepers *keepers.AppKeepers, expectedCoinsUsedInUpgradeHandler sdk.Coins, lastPoolID uint64) {
+			func(ctx sdk.Context, keepers *keepers.AppKeepers, expectedCoinsUsedInUpgradeHandler sdk.Coins, lastPoolID uint64, clPoolTwapRecordPreUpgrade []types.TwapRecord) {
```

Comment:
Consider avoiding passing in `clPoolTwapRecordPreUpgrade` here and returning them from pre-upgrade logic. Instead, use the pool id similar to the other pool